lib/navigation/model/route_paths.dart

```diff
@@ -0,0 +1,347 @@
+
+  @override
+  Widget get page => const ClassFeedbackChecklist();
+}
+
+class ClassFeedbackViewPath extends RoutePath {
+  ClassFeedbackViewPath(this.id) : super(ClassFeedbackView.routeName);
+
+  final String id;
+
+  // TODO(RazvanRotaru): Retrieve [ClassHeader] for [ClassFeedbackView] by id
+  @override
+  Widget get page => const ClassFeedbackView();
+}
+
+class ClassViewPath extends RoutePath {
+  ClassViewPath(this.id) : super(ClassView.routeName);
+
+  final String id;
+
+  // TODO(RazvanRotaru): retrieve [ClassHeader] for [ClassView] by id
+  @override
+  Widget get page => const ClassView();
+}
+
+class ClassesPagePath extends RoutePath {
+  ClassesPagePath() : super(ClassesPage.routeName);
+
+  @override
+  Widget get page => const ClassesPage();
+}
+
+class AddClassesPath extends RoutePath {
+  AddClassesPath() : super(AddClassesPage.routeName);
+
+  @override
+  Widget get page => const AddClassesPage();
+}
+
+class EventViewPath extends RoutePath {
+  EventViewPath(this.id) : super('${EventView.routeName}?id=$id');
+
+  final String id;
+
+  // TODO(RazvanRotaru): retrieve [Event] for [EventView] by id
+  @override
+  Widget get page => EventView();
+}
+
+class AddEventPath extends RoutePath {
+  AddEventPath() : super(AddEventView.routeName);
+
+  @override
+  Widget get page => const AddEventView();
+}
+
+class UnknownPath extends RoutePath {
+  UnknownPath() : super('');
+
+  // TODO(RazvanRotaru): Use [ErrorPage] instead of plain [Text]
+  @override
+  Widget get page => const Text('404! Unknown location');
+}
+
+// TODO(RazvanRotaru): auto-generate PathFactory
+class PathFactory {
+  static RoutePath from(Uri uri) {
+    if (uri.pathSegments.isEmpty) {
+      return RootPath();
+    }
+
+    // final String path = uri.path;
+    switch (uri.path) {
+      case HomePage.routeName:
+        return HomePath();
+      case TimetablePage.routeName:
+        return TimetablePath();
+      case PortalPage.routeName:
+        return PortalPath();
+      case SettingsPage.routeName:
+        return SettingsPath();
+      case FilterPage.routeName:
+        return FilterPath();
+      case LoginView.routeName:
+        return LoginPath();
+      case SignUpView.routeName:
+        return SignUpPath();
+      case FaqPage.routeName:
+        return FaqPath();
+      case NewsFeedPage.routeName:
+        return NewsFeedPath();
+      case RequestPermissionsPage.routeName:
+        return RequestPermissionsPath();
+      case PeoplePage.routeName:
+        return PeoplePath();
+      case PersonView.routeName:
+        final name = uri.queryParameters['name'];
+        print('Factory-Person: $name');
+        return ProfilePath(name);
+      case EditProfilePage.routeName:
+        return EditProfilePath();
+      case WebsiteView.routeName:
+        final id = uri.queryParameters['id'];
+        return WebsiteViewPath(id);
+
+      // this case might be the reason why auto-generation won't really work
```

Comment:
Would it work if we created a new widget like AddWebsiteView and used that? (and the build method would just return WebsiteView with the right params)

Reply:
I believe it would but do I look like I know how to do that? :)) 
I was thinking that each page should know this information about itself (the RouteInformation data is the only data that is missing from a page). But in order for this to work fine, the arguments passed to the constructor should be only what is relevant for a URI (if this doesn't make sense I'm sorry. it's late).

I guess we could ditch the autogeneration for the time being in order to settle the direction of the navigation :-??

Reply:
my god is this gross
![image](https://user-images.githubusercontent.com/15521900/134585693-df770e67-6a2a-4103-984f-21b1cfb73973.png)